### PR TITLE
test with highest and lowest python version (GSI-452)

### PR DESCRIPTION
**Pull request by KerstenBreuer**
```
Run tests with highest and lowest supported python version.
It is assumed that all dependencies as listed in ./lock/requirements-dev.txt
work for both python versions.
```

### CODE REVIEW

File: .pyproject_generation/pyproject_custom.toml [hunk 1]
```diff
@@ -22,3 +22,20 @@ all = ["hexkit[test]"]
 
 [project.urls]
 Repository = "https://github.com/ghga-de/hexkit"
+
+[tool.tox]
+legacy_tox_ini = """
```
Comment (Cito) [inline .pyproject_generation/pyproject_custom.toml:27]: I had hoped they would support toml natively in tox 4, but looks like we still need to do it hat way.

**Review by Cito — APPROVED** [review]
Nice, looks very simple in hindsight.

File: lock/requirements-dev.in [hunk 2]
```diff
@@ -7,3 +7,4 @@
 cryptography >= 41
 func-timeout >= 4.3.5
 pytest-asyncio >= 0.23.0
+tox >= 4.11.4
```
Comment (Cito) [inline lock/requirements-dev.in:10]: tox 4.12 has just been released :)

File: .pyproject_generation/pyproject_custom.toml (hunk 1 above)
Comment (KerstenBreuer) [reply .pyproject_generation/pyproject_custom.toml:27]: Yeah, seems so

File: lock/requirements-dev.in (hunk 2 above)
Comment (KerstenBreuer) [reply lock/requirements-dev.in]: Great, I updated.

**Review by Cito — APPROVED** [review]

**Pull request merged**